src/databricks/sql/client.py

```diff
@@ -519,36 +608,60 @@ def execute(
     ) -> "Cursor":
         """
         Execute a query and wait for execution to complete.
-        Parameters should be given in extended param format style: %(...)<s|d|f>.
-        For example:
-            operation = "SELECT * FROM table WHERE field = %(some_value)s"
-            parameters = {"some_value": "foo"}
-            Will result in the query "SELECT * FROM table WHERE field = 'foo' being sent to the server
+
+        The parameterisation behaviour of this method depends on which parameter approach is used:
+            - With INLINE mode (default), parameters are rendered inline with the query text
+            - With NATIVE mode, parameters are sent to the server separately for binding
+
+        This behaviour is controlled by the `use_inline_params` argument passed when building a connection.
+
+        The syntax for these approaches is different:
+
+        If the connection was instantiated with use_inline_params=False, then parameters
+        should be given in PEP-249 `named` paramstyle like :param_name
+
+        If the connection was instantiated with use_inline_params=True (default), then parameters
+        should be given in PEP-249 `pyformat` paramstyle like %(param_name)s
+
+        ```python
+        inline_operation = "SELECT * FROM table WHERE field = %(some_value)s"
+        native_operation = "SELECT * FROM table WHERE field = :some_value"
+        parameters = {"some_value": "foo"}
+        ```
+
+        Both will result in the query equivalent to "SELECT * FROM table WHERE field = 'foo'
+        being sent to the server
+
         :returns self
         """
-        if parameters is None:
-            parameters = []
 
-        elif not Connection.server_parameterized_queries_enabled(
-            self.connection.protocol_version
-        ):
-            raise NotSupportedError(
-                "Parameterized operations are not supported by this server. DBR 14.1 is required."
-            )
+        if parameters:
+            param_approach = self._determine_parameter_approach()
```

Comment:
I'm open to feedback from reviewers. Does it make sense to remove the conditional on line 638 and then update `_determine_parameter_approach` so that it can sense that no parameters were provided and return `ParameterApproach.NONE`?

Reply:
I have applied this change.